### DISCUSSION TITLE
Lowering resolution/not rendering offscreen code blocks to improve performance in editor

### DISCUSSION THREAD

**Discussion by Zyphiria** [ScratchAddons/ScratchAddons#7683] [Ideas]
When there's a lot of code blocks in one sprite in the Scratch editor, just having that sprite open puts a ton of lag on slower systems. Even if the project isn't running, just having a sprite with 500 code blocks open will slow down my editor to like 10 frames/s. (I have a slower pc to be fair)
No idea how the Scratch editor works performance-wise. I'm assuming that if it has to render more code blocks at once, it slows down. If there was an addon for code block culling or lowering their resolution, that would like, probably help my twelve year old processor run Scratch better?

**Comment by DNin01** [comment]
Scratch isn't using the latest version of Blockly, Google's block coding technology, and TurboWarp already runs quite a bit smoother in the code editor, so obviously there's room for improvement.

These types of addons are also worthy of being on by default, and maybe even set to the side if we ever group optimization/tweak addons into their own section. (Related: #2617, #7152)
